### PR TITLE
docs: add command reference

### LINKED ISSUE #855: Document all the flags supported by the driver
/kind documentation

https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/40a680657ca5f44c2893eb2098fd94b97b868c2a/cmd/secrets-store-csi-driver/main.go#L54-L69

- This will make it easier for announce flag deprecations
- Gives a better understanding of what the flag is and when to use it

### PR DESCRIPTION
**Pull request by umeshkumhar**
**What this PR does / why we need it**: This is raised to add a new subpage under topics around command reference

**Which issue(s) this PR fixes**:
Fixes #855 

**Special notes for your reviewer**: 
- Do let me know if **Command Reference** title works else we can update with **Command Line Arguments** or **Configuration Options**

### CODE REVIEW

File: docs/book/src/topics/command-reference.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+| `--nodeid`                           | Node ID                                                                |                                             |
+| `--log-format-json`                  | Set log formatter to json                                              | `false`                                       |
+| `--provider-volume`                  | Volume path for provider                                               | `/etc/kubernetes/secrets-store-csi-providers` |
+| `--additional-provider-volume-paths` | Comma separated list of additional paths to communicate with providers | `/var/run/secrets-store-csi-providers`        |
+| `--metrics-addr`                     | The address the metric endpoint binds to                               | `:8095`                                       |
+| `--enable-secret-rotation`           | Enable secret rotation feature [alpha]                                 | `false`                                       |
+| `--rotation-poll-interval`           | Secret rotation poll interval duration                                 |                                             |
```
Comment (nilekhc) [inline docs/book/src/topics/command-reference.md:19]: default rotation poll interval is `2m`

File: docs/book/src/topics/command-reference.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Command Reference
+
+The Secrets Store CSI Driver can be provided with the following command line arguments.
```
Comment (aramase) [inline docs/book/src/topics/command-reference.md:3]:
```suggestion
The `secrets-store` container in the DaemonSet can be configured using the following command line arguments:
```

**Review by aramase — COMMENTED** [review]
Thanks for the PR @umeshkumhar! 

overall lgtm, added one minor comment. Could you also [squash](https://kubernetes.io/docs/contribute/new-content/open-a-pr/#squashing-commits) to single commit?

**Comment by nilekhc** [comment]
> Thanks for the PR @umeshkumhar!
> 
> overall lgtm, added one minor comment. Could you also [squash](https://kubernetes.io/docs/contribute/new-content/open-a-pr/#squashing-commits) to single commit?

/hold

**Review by aramase — CHANGES_REQUESTED** [review]
The changes in `Makefile`, `docker/Dockerfile` and `docker/Makefile` are already in the main branch. Looks like this PR needs to be rebased so those changes aren't being added again as part of the PR. Could you rebase the PR on top of the main branch?

**Pull request merged**